libs/cmdio/logger.go

```diff
@@ -74,24 +75,66 @@ func LogError(ctx context.Context, err error) {
 	})
 }
 
-func Ask(ctx context.Context, question string) (bool, error) {
+func Ask(ctx context.Context, question, defaultVal string) (string, error) {
 	logger, ok := FromContext(ctx)
 	if !ok {
 		logger = Default()
 	}
-	return logger.Ask(question)
+	return logger.Ask(question, defaultVal)
 }
 
-func (l *Logger) Ask(question string) (bool, error) {
+func AskYesOrNo(ctx context.Context, question string) (bool, error) {
+	logger, ok := FromContext(ctx)
+	if !ok {
+		logger = Default()
+	}
+	return logger.AskYesOrNo(question)
+}
+
+func (l *Logger) Ask(question string, defaultVal string) (string, error) {
+	if l.Mode == flags.ModeJson {
+		return "", fmt.Errorf("question prompts are not supported in json mode")
+	}
+
+	// Add default value to question prompt.
+	if defaultVal != "" {
+		question += fmt.Sprintf(` [%s]`, defaultVal)
+	}
+	question += `: `
+
+	// print prompt
+	_, err := l.Writer.Write([]byte(question))
+	if err != nil {
+		return "", err
+	}
+
+	// read user input. Trim new line characters
+	ans, err := l.Reader.ReadString('\n')
+	if err != nil {
+		return "", err
+	}
+	ans = strings.Trim(ans, "\n\r")
+
+	// Return default value if user just presses enter
+	if ans == "" {
+		return defaultVal, nil
+	}
+	return ans, nil
+}
+
+func (l *Logger) AskYesOrNo(question string) (bool, error) {
```

Comment:
Could you reuse `l.Ask` here instead? It will then look like

```
ans, error := l.Ask(question, "y/n")
if ans == "y\n" {
	return true, nil
} else {
	return false, nil
}
```

Reply:
done